calendar/modules/listener/watch_scheduler.bal

```diff
@@ -14,20 +14,46 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import ballerina/lang.runtime;
 import ballerina/task;
 import ballerina/time;
 
 class Job {
     *task:Job;
     private Listener 'listener;
+    private int retryCount = 0;
+    private int retryScheduleCount = 0;
 
     isolated function init(Listener 'listener) {
         self.'listener = 'listener;
     }
 
     public isolated function execute() {
-        checkpanic self.'listener.registerWatchChannel();
-        checkpanic self.scheduleNextChannelRenewal();
+        error? err = self.'listener.registerWatchChannel();
+        if (err is error) {
```

Comment:
Shall we print a WARN log saying `Could not register watch channel`. Something like 
`log.warn("Could not register watch channel" +  error.toString());`